bundles/org.openhab.binding.myuplink/src/main/java/org/openhab/binding/myuplink/internal/provider/ChannelFactory.java

```diff
@@ -0,0 +1,236 @@
+    private final ChannelTypeRegistry channelTypeRegistry;
+
+    @Activate
+    public ChannelFactory(@Reference MyUplinkChannelTypeProvider channelTypeProvider,
+            @Reference ChannelTypeRegistry channelTypeRegistry) {
+        this.channelTypeProvider = channelTypeProvider;
+        this.channelTypeRegistry = channelTypeRegistry;
+    }
+
+    public Channel createChannel(ThingUID thingUID, JsonObject channelData) {
+        final var channelId = Utils.getAsString(channelData, JSON_KEY_CHANNEL_ID, GENERIC_NO_VAL);
+        final var label = Utils.getAsString(channelData, JSON_KEY_CHANNEL_LABEL, GENERIC_NO_VAL);
+        final var unit = Utils.fixUnit(Utils.getAsString(channelData, JSON_KEY_CHANNEL_UNIT, ""));
+        final var strVal = Utils.getAsString(channelData, JSON_KEY_CHANNEL_STR_VAL, GENERIC_NO_VAL);
+        final var writable = Utils.getAsBool(channelData, JSON_KEY_CHANNEL_WRITABLE, Boolean.FALSE);
+        final var enumValues = Utils.getAsJsonArray(channelData, JSON_KEY_CHANNEL_ENUM_VALUES);
+        final var minValue = Utils.getAsBigDecimal(channelData, JSON_KEY_CHANNEL_MIN);
+        final var maxValue = Utils.getAsBigDecimal(channelData, JSON_KEY_CHANNEL_MAX);
+        final var stepValue = Utils.getAsBigDecimal(channelData, JSON_KEY_CHANNEL_STEP);
+
+        ChannelTypeUID channelTypeUID = null;
+        if (enumValues.isEmpty()) {
+            if (!writable) {
+                channelTypeUID = determineStaticChannelTypeUID(unit, strVal.contains(JSON_VAL_DECIMAL_SEPARATOR));
+            } else {
+                channelTypeUID = getOrBuildNumberChannelType(channelId, unit, minValue, maxValue, stepValue);
+            }
+        } else {
+            channelTypeUID = determineEnumChannelTypeUID(channelId, enumValues, writable);
+        }
+
+        final var channelUID = new ChannelUID(thingUID, channelId);
+        final var acceptedType = determineAcceptedType(channelTypeUID, unit);
+        final var builder = ChannelBuilder.create(channelUID).withLabel(label).withDescription(label)
+                .withType(channelTypeUID).withAcceptedItemType(acceptedType);
+
+        if (writable) {
+            var props = new HashMap<String, String>();
+            props.put(PARAMETER_NAME_VALIDATION_REGEXP, DEFAULT_VALIDATION_EXPRESSION);
+            builder.withProperties(props);
+        }
+
+        return builder.build();
+    }
+
+    String determineAcceptedType(ChannelTypeUID channelTypeUID, String unit) {
+        if (channelTypeUID.getId().equals(CHANNEL_TYPE_RW_SWITCH)) {
+            return CoreItemFactory.SWITCH;
+        } else if (unit.isEmpty()) {
+            return CoreItemFactory.NUMBER;
+        } else {
+            Unit<?> parsedUnit = UnitUtils.parseUnit(unit);
+            String dimension = parsedUnit == null ? null : UnitUtils.getDimensionName(parsedUnit);
+
+            if (dimension == null || dimension.isEmpty()) {
+                logger.info("Could not parse unit: '{}'", unit);
```

Comment:
I see info level is used in more please, please check https://www.openhab.org/docs/developer/guidelines.html#f-logging
Most cases are probably warn.

```suggestion
                logger.warn("Could not parse unit: '{}'", unit);
```

Reply:
In the past I used "warn" more often but I got feedback that "warn" should only be used in case of severe issues. Maybe the guidelines have changed.

Reply:
The guidelines regarding logging have not been changed for some time. Hard to tell about the earlier feedback. I can only comment on this case now.